zerver/lib/actions.py

```diff
@@ -1499,7 +1500,13 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
             event['sender_queue_id'] = message['sender_queue_id']
         send_event(message['realm'], event, users)
 
-        if url_embed_preview_enabled(message['message']) and links_for_embed:
+        whitelisted_urls = get_whitelisted_urls(links_for_embed)
+        if not url_embed_preview_enabled(message['message']) and whitelisted_urls:
```

Comment:
I think the two conditionals here can be simplified to the following? :thinking:

```python
if not url_embed_preview_enabled(message['message']):
    links_for_embed = get_whitelisted_urls(links_for_embed)

if links_for_embed:
     event_data = {...}
```

Also, thinking more about this, I'm wondering if we even need to check for 1) whether url_embed_preview_enabled and 2) if the URL is whitelisted. We adding URLs to the `links_for_embed` set, only when the required conditions are met -- i.e., either the flag is turned on, or the URL is white listed. I think it should just be sufficient to check if `links_for_embed` is not empty? If this indeed makes sense, I would first make a preparatory refactor commit changing the `if url_embed_preview_enabled(message['message']) and links_for_embed` conditional to `if links_for_embed` before the whitelisted domains change.